flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingExecutor.java

```diff
@@ -222,19 +223,30 @@ private Map<JobVertexID, ScalingSummary> computeScalingSummary(
             Map<JobVertexID, SortedMap<Instant, ScalingSummary>> scalingHistory) {
 
         var out = new HashMap<JobVertexID, ScalingSummary>();
+        var excludeVertexIdList = conf.get(AutoScalerOptions.VERTEX_EXCLUDE_IDS);
         evaluatedMetrics.forEach(
                 (v, metrics) -> {
-                    var currentParallelism =
-                            (int) metrics.get(ScalingMetric.PARALLELISM).getCurrent();
-                    var newParallelism =
-                            jobVertexScaler.computeScaleTargetParallelism(
-                                    resource,
-                                    conf,
+                    if (excludeVertexIdList.contains(v.toHexString())) {
+                        LOG.debug(
+                                "Vertex {} is part of `vertex.exclude.ids` config, Ignoring it for scaling",
+                                v);
```

Comment:
This still requires more work to ensure the feature works for non-sink vertices. If you ignore a non-sink vertex, the downstream operators from it might be scaled based on the target rates calculated for the ignored vertex. We need to make sure to set the target rates based on the current rate for the ingored vertex. Here: https://github.com/apache/flink-kubernetes-operator/blob/c6cd7cbb83c5257960daebe65214035a6732ad0d/flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricEvaluator.java#L91

Reply:
got the issue, thanks, fixing it